cmd/juju/commands/ssh_container.go

```diff
@@ -187,6 +188,14 @@ func (c *sshContainer) cleanupRun() {
 const charmContainerName = "charm"
 
 func (c *sshContainer) resolveTarget(target string) (*resolvedTarget, error) {
+	if c.modelName == environsbootstrap.ControllerModelName {
+		// TODO(caas): change here to controller unit tag once we refactored controller to an application.
+		if target != "0" {
```

Comment:
This breaks applications deployed in the controller model, which should still work.

This change would be sufficient.
```
	if c.modelName == environsbootstrap.ControllerModelName && target == "0" {
		// TODO(caas): change here to controller unit tag once we refactored controller to an application.
		return &resolvedTarget{entity: fmt.Sprintf("%s-%s", environsbootstrap.ControllerModelName, target)}, nil
	}
```

Reply:
That's right. Even it's not recommended to deploy a workload to the controller model, we can stop ppl from doing that.